Build via Docker image fails: `fatal error: cblas.h: No such file or directory`

After fixing the issue from #313 (via #314), `make` gets to another problem:

```
[ 87%] Building CXX object thirdparty/milvus/internal/core/thirdparty/knowhere/knowhere-build/CMakeFiles/knowhere.dir/src/common/thread/thread.cc.o
cd /workspace/python/build/bdist.linux-x86_64/build_milvus/thirdparty/milvus/internal/core/thirdparty/knowhere/knowhere-build && /opt/rh/gcc-toolset-12/root/usr/bin/c++ -DANTLR4CPP_STATIC -DCURL_STATICLIB=1 -DFMT_HEADER_ONLY -DFMT_HEADER_ONLY=1 -DFOLLY_HAVE_DWARF -DFOLLY_HAVE_ELF -DHAVE_CPP_STDLIB -DKNOWHERE_VERSION=v2.5.6 -DKNOWHERE_WITH_LIGHT -DLZMA_API_STATIC -DMILVUS_LITE -DNOT_COMPILE_FOR_SWIG -DSODIUM_STATIC -Dknowhere_EXPORTS -I/workspace/python/build/bdist.linux-x86_64/build_milvus/thirdparty/milvus/internal/core/thirdparty/knowhere/knowhere-src/thirdparty/hnswlib -I/workspace/python/build/bdist.linux-x86_64/build_milvus/thirdparty/milvus/internal/core/thirdparty/knowhere/knowhere-src/thirdparty/faiss -I/workspace/python/build/bdist.linux-x86_64/build_milvus/thirdparty/milvus/internal/core/thirdparty/knowhere/knowhere-src/src -I/workspace/python/build/bdist.linux-x86_64/build_milvus/thirdparty/milvus/internal/core/thirdparty/knowhere/knowhere-src/include -isystem /root/.conan/data/boost/1.82.0/_/_/package/698576ce8efdf34185d38fc8dc06558b770ac2a6/include -isystem /root/.conan/data/glog/0.6.0/_/_/package/1646d92186d2eddfba2f359f74502188a8875d08/include -isystem /root/.conan/data/gflags/2.2.2/_/_/package/a0d353f0c840e11bc3b4b6a6508764ddb2f83a46/include -isystem /root/.conan/data/libunwind/1.7.2/_/_/package/834ed377e7c47e86a4020acdf4a045bf0c7ca8ca/include -isystem /root/.conan/data/xz_utils/5.4.5/_/_/package/2a19826344ff00be1c04403f2f8e7008ed3a7cc6/include -isystem /root/.conan/data/zlib/1.2.13/_/_/package/2a19826344ff00be1c04403f2f8e7008ed3a7cc6/include -isystem /root/.conan/data/nlohmann_json/3.11.2/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include -isystem /root/.conan/data/prometheus-cpp/1.1.0/_/_/package/9787ccaba251d035da8d731ecb9293b605e6e0e8/include -isystem /root/.conan/data/libcurl/7.86.0/_/_/package/2b86e3b2b65733b8f0a17ebbcea32755f5c16632/include -isystem /root/.conan/data/fmt/9.1.0/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include -isystem /root/.conan/data/folly/2023.10.30.09/milvus/dev/package/3df752894b26fc56e8bef366f933de99be775aa5/include -isystem /root/.conan/data/bzip2/1.0.8/_/_/package/3cfc45772763dad1237052f26c1fe8b2bae3f7d2/include -isystem /root/.conan/data/double-conversion/3.2.1/_/_/package/9428c85bc64cf85711c31a116b265ea0dd5903a4/include -isystem /root/.conan/data/libevent/2.1.12/_/_/package/fc397aedff33f10a79b59bd27bbf707923621ce2/include -isystem /root/.conan/data/openssl/1.1.1t/_/_/package/2a19826344ff00be1c04403f2f8e7008ed3a7cc6/include -isystem /root/.conan/data/lz4/1.9.4/_/_/package/2a19826344ff00be1c04403f2f8e7008ed3a7cc6/include -isystem /root/.conan/data/snappy/1.1.9/_/_/package/20e27b6454facc8034c95796240a5425be6f8e87/include -isystem /root/.conan/data/zstd/1.5.4/_/_/package/1b177e416597e90db2922415c5ce0fe8e7df6e75/include -isystem /root/.conan/data/libsodium/cci.20220430/_/_/package/e1862765807671e4789b8055ed202d407ab0b882/include -isystem /root/.conan/data/libdwarf/0.9.1/_/_/package/5179b38d255e28b4e2c7b248232d30439b996b73/include -isystem /root/.conan/data/libiberty/9.1.0/_/_/package/2a19826344ff00be1c04403f2f8e7008ed3a7cc6/include -msse4.2 -Wall -fPIC -std=gnu++17 -msse4.2 -Wall -fPIC -std=gnu++17 -O3 -Wall -fPIC  -fopenmp -O3 -DNDEBUG -std=gnu++17 -fPIC -fopenmp -MD -MT thirdparty/milvus/internal/core/thirdparty/knowhere/knowhere-build/CMakeFiles/knowhere.dir/src/common/thread/thread.cc.o -MF CMakeFiles/knowhere.dir/src/common/thread/thread.cc.o.d -o CMakeFiles/knowhere.dir/src/common/thread/thread.cc.o -c /workspace/python/build/bdist.linux-x86_64/build_milvus/thirdparty/milvus/internal/core/thirdparty/knowhere/knowhere-src/src/common/thread/thread.cc
In file included from /workspace/python/build/bdist.linux-x86_64/build_milvus/thirdparty/milvus/internal/core/thirdparty/knowhere/knowhere-src/src/common/thread/thread.cc:21:
/workspace/python/build/bdist.linux-x86_64/build_milvus/thirdparty/milvus/internal/core/thirdparty/knowhere/knowhere-src/include/knowhere/comp/thread_pool.h:22:10: fatal error: cblas.h: No such file or directory
   22 | #include <cblas.h>
      |          ^~~~~~~~~
compilation terminated.
gmake[2]: *** [thirdparty/milvus/internal/core/thirdparty/knowhere/knowhere-build/CMakeFiles/knowhere.dir/build.make:191: thirdparty/milvus/internal/core/thirdparty/knowhere/knowhere-build/CMakeFiles/knowhere.dir/src/common/thread/thread.cc.o] Error 1
```

Looks like `openblas-devel` installs headers to `/usr/include/openblas`.

--- a/thirdparty/knowhere.patch
+++ b/thirdparty/knowhere.patch
@@ -12,3 +12,37 @@ index b7870adf..0d8fc16c 100644
      src/index/index_node_data_mock_wrapper.cc
      src/index/index_static.cc
      src/index/index.cc
+diff --git a/cmake/libs/libfaiss.cmake b/cmake/libs/libfaiss.cmake
+index 2756fb57..635358a3 100644
+--- a/cmake/libs/libfaiss.cmake
++++ b/cmake/libs/libfaiss.cmake
+@@ -113,6 +113,7 @@ endif()
+ 
+ if(LINUX)
+   set(BLA_VENDOR OpenBLAS)
++  find_path(BLAS_INCLUDE_DIRS cblas.h PATH_SUFFIXES openblas)
+ endif()
+ 
+ if(APPLE)
+@@ -168,6 +169,7 @@ if(__X86_64)
+     faiss PUBLIC OpenMP::OpenMP_CXX ${BLAS_LIBRARIES} ${LAPACK_LIBRARIES}
+                  faiss_avx2 faiss_avx512 knowhere_utils)
+   target_compile_definitions(faiss PRIVATE FINTEGER=int)
++  target_include_directories(faiss PUBLIC ${BLAS_INCLUDE_DIRS})
+ endif()
+ 
+ if(__AARCH64)
+@@ -190,6 +192,7 @@ if(__AARCH64)
+   target_link_libraries(faiss PUBLIC OpenMP::OpenMP_CXX ${BLAS_LIBRARIES} ${LAPACK_LIBRARIES}
+                                      knowhere_utils)
+   target_compile_definitions(faiss PRIVATE FINTEGER=int)
++  target_include_directories(faiss PUBLIC ${BLAS_INCLUDE_DIRS})
+ endif()
+ 
+ if(__PPC64)
+@@ -217,4 +220,5 @@ if(__PPC64)
+   target_link_libraries(faiss PUBLIC OpenMP::OpenMP_CXX ${BLAS_LIBRARIES} ${LAPACK_LIBRARIES}
+                                       knowhere_utils)
+   target_compile_definitions(faiss PRIVATE FINTEGER=int)
++  target_include_directories(faiss PUBLIC ${BLAS_INCLUDE_DIRS})
+ endif()

